next/pages/_app.tsx

```diff
@@ -1,10 +1,21 @@
 import "styles/globals.css";
+import React from 'react';
 
 import { AppProps } from "next/app";
 import { appWithTranslation } from "next-i18next";
 
+const default_langs = {
```

Comment:
I think this is a bit of a problem, because then in our instructions we would need to  tell people to come here and add the set of languages, which I think we should avoid.

Reply:
for this react side of things, I think it would make sense to talk for example with @juhov or some other people that have actually implemented these kind of frontend sites, so we can _reuse_ their experience as well (since this is about reusability)

Reply:
Unfortunately I don't have experience about language's as nelonen.fi is Finnish only.

There is some prior art from the next-drupal people in the Umami example. In their setup the languages among some other things are defined in a [site.config.js](https://github.com/chapter-three/next-drupal/blob/main/examples/example-umami/site.config.js) file.

Reply:
The problem is slug part translation.
In Drupal out of the box you have the pattern, that alias follows the title of the article, thus Eng and Fi versions of the same article will have different URLs.
In Umami example, both Spanish and English URLs are identical, therefore they can use in their example
<Link href={asPath} locale={locale} passHref>
and we can't do that. Simply put locale in front of your [slug] doesn't help. We need to fetch translated paths and pass them to LangSwitcher.

Reply:
@ekondareva I understand, we can't get their language switcher component, I propose we get their idea of that site.config file, and have there the settings for languages in one place.

We could have: 
* name of the language
* langcode
* homepage for the language ( "/" for english, "/sv" for swedish), etc

then we can feed that info info the query that you did , and maybe even use the language name in the language switcher? 

would this work you think?

Reply:
I'm thinking... if the jsonapi response for the node could include the paths of its translation then we would not need to do additional queries in the frontend.

I will look into whether we can add a field like that.

Reply:
^ so, it seems it might be possible, but if we want to switch to graphql later we would need to do this work twice, so for now I propose: 
* let's try adopting that `site.config` file
* let's go with the multiple queries

Reply:
My two cents: If the options for getting the translated paths are
A) customize JSON API in Drupal or
B) do one extra query in Next.js 

I favor B, because translatePathFromContext() is and documented plus it makes Next.js developer less dependent of Drupal. And not having custom stuff in Drupal is one less thing to maintain.